src/qibo/backends/photonic_strong_simulation.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+from perceval import (
+    BS,
+    BasicState,
+    Experiment,
+    Processor,
+    RemoteProcessor,
+)
+from perceval import __version__ as pcvl_version
+from perceval import (
+    probs_to_samples,
+)
+from perceval.algorithm import Sampler
+
+from qibo import Circuit, __version__
+from qibo.backends.numpy import NumpyBackend, NumpyMatrices
+from qibo.gates import H, PhotonicGate
+from qibo.measurements import MeasurementResult
+
+from .modality import Modality
+
+
+class LOQCMatrices:
+    def __init__(self, dtype):
+        self.dtype = dtype
+
+    def _cast(self, x, dtype):
+        if isinstance(x, list):
+            return np.array(x, dtype=dtype)
+        return x.astype(dtype)
+
+    def BS(self, theta):
+        cos = np.cos(theta / 2.0) + 0j
+        isin = -1j * np.sin(theta / 2.0)
+        return self._cast([[cos, isin], [isin, cos]], dtype=self.dtype)
```

Comment:
Like the Hadamard gate that is already defined in `qibo`, this gate seems to be the same as the `gates.RX`.